### PR TITLE
Convert boskos deployment to a statefulset

### PR DESCRIPTION
**Pull request by msau42**
I only tested creating the statefulset + volumes to make sure the names are correct and it mounts properly.  But I haven't done any e2e testing to determine if there's anything else that needs to be updated.

### CODE REVIEW

**Comment by msau42** [comment]
For #4571

File: boskos/deployment.yaml [hunk 1]
```diff
@@ -35,24 +38,24 @@ kind: PersistentVolumeClaim
...
   accessModes:
     - ReadWriteOnce
   resources:
     requests:
       storage: 1Gi
 ---
 # Start of Deployment
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: apps/v1beta1
```
Comment (foxish) [inline boskos/deployment.yaml]: Comment above needs update

File: boskos/deployment.yaml [hunk 2]
```diff
@@ -35,24 +38,24 @@ kind: PersistentVolumeClaim
 metadata:
   labels:
     app: boskos
-  name: boskos-storage
+  name: boskos-volume-boskos-0
```
Comment (foxish) [inline boskos/deployment.yaml]: We don't need this anymore because the StatefulSet controller will stamp out a PVC

**Comment by msau42** [comment]
Updated.  The filename itself should also change but I wasn't sure if that would break any scripts.

**Comment by foxish** [comment]
lgtm. Thanks!

**Comment by krzyzacy** [comment]
(deployed - looks working except that seems the cache saved inside previous pv is wiped? but seems it regenerated fine)

**Comment by msau42** [comment]
Hm that shouldn't happen since the PV reclaim policy is retain.

**Comment by msau42** [comment]
@krzyzacy if it works for you, then go ahead and merge this when you're ready.

**Comment by krzyzacy** [comment]
/lgtm

I'll watch the cache issue next time we redeploy. Huge thanks!

**Pull request merged**